### PR TITLE
Added Data Field options as a combobox

### PR DESCRIPTION
**Pull request by JonTheNiceGuy**
* Added datalist to the "new" field.
* Added link to revealjs documentation.

### CODE REVIEW

**Comment by JonTheNiceGuy** [comment]
This should, in theory, resolve #6

File: js/edit-slide-admin.js [hunk 1]
```diff
@@ -61,7 +61,27 @@ jQuery( document ).ready( function( $ ) {
 	$( '#slides' ).on( 'click.add-data', '.button.add-data', function( e ) {
 		var table_body = $(this).closest( 'table.slide-data-attributes-table' ).find( 'tbody' );
 		var slide_index = $(this).closest( '.stuffbox' ).find( 'input[name="slide-index"]' ).val();
-		var data_row = '<tr><td class="left newdataleft"><input type="text" name="slide-data[' + slide_index + '][]"></td><td><input type="text" name="slide-data-value[' + slide_index + '][]"></td></tr>';
+		var row_id = Math.floor(Math.random() * 100000000)
+		var data_row = '<tr>';
+		data_row += '<td class="left newdataleft">';
+		data_row += '<input type="text" name="slide-data[' + slide_index + '][]" list="slide-data-' + row_id + '">';
+		data_row += '<datalist id="slide-data-' + row_id + '">';
```
Comment (aaroncampbell) [inline js/edit-slide-admin.js]: Can we just add a single datalist outside of this (or add it only if it doesn't already exist somewhere) and then reuse it from each input? My understanding is that something like this is perfectly acceptable:
```
<datalist id="mylist">
	<option value="one">
	<option value="two">
	<option value="three">
</datalist>
<input list="mylist" name="test-one">
<input list="mylist" name="test-two">
<input list="mylist" name="test-three">
```

**Review by aaroncampbell — CHANGES_REQUESTED** [review]

File: js/edit-slide-admin.js (hunk 1 above)
Comment (JonTheNiceGuy) [reply js/edit-slide-admin.js]: Fair point. I'll get some time to take a look at fixing that up tonight or tomorrow night. Good spot though!

Comment (JonTheNiceGuy) [reply js/edit-slide-admin.js]: Took a little longer than expected. PR updated.

**Comment by JonTheNiceGuy** [comment]
Hi @aaroncampbell, just wanted to check, do you need me to redo anything with this, or is it just something that won't be in scope any more? Fine either way, just wanted to check! I'm happy to help out with stuff if you need assistance in the background on this?

**Comment by aaroncampbell** [comment]
Sorry, I think this was inadvertently closed when I renamed the master branch to main. Please feel free to reopen for the new branch. Glancing at the code, it looks like there is a `row_id` variable that isn't being used?

**Pull request closed without merging**